### PR TITLE
Update integration test instructions

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * N/A I've added or updated automated unit tests to verify correctness of my new code.
   * N/A I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * N/A I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Updates CONTRIBUTING.md with latest instructions for running integration tests locally and updates the integration test account setup script to make it runnable and idempotent.

### CODE REVIEW

File: tests_integration/scripts/integration_account_setup.sql [hunk 1]
```diff
@@ -63,7 +63,7 @@ CREATE OR REPLACE NETWORK RULE snowflake_docs_network_rule
 
 CREATE OR REPLACE SECRET test_secret
   TYPE = GENERIC_STRING
---   SECRET_STRING = ''; -- provide password
+  SECRET_STRING = 'test';
```
Comment (sfc-gh-fcampbell) [inline tests_integration/scripts/integration_account_setup.sql:66]: i'm not sure why this was commented out originally, without this param the query fails

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,44 +65,48 @@ or by running `pytest` inside activated environment.
...
+- `HOST="<your host here>"`
+- `ACCOUNT="<your account here>"`
+- `USER="snowcli_test"`
+- `PRIVATE_KEY_PATH="~/.ssh/snowcli_test_rsa_key.p8"`
 
+### Invoking tests
+To invoke tests, either `export` the above environment variables in your shell/bashrc/zshrc or specify them when invoking the test command:
 ```bash
-tests_integration/scripts/integration_account_setup.sql
+SNOWFLAKE_CONNECTIONS_INTEGRATION_HOST="<your host here>" \
```
Comment (sfc-gh-melnacouzi) [inline CONTRIBUTING.md:105]: how about warehouse, role, and authenticator (SNOWFLAKE_JWT)?

Comment (sfc-gh-fcampbell) [reply CONTRIBUTING.md:105]: yeah i thought they wouldn't be required but they are, will update

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -65,44 +65,48 @@ or by running `pytest` inside activated environment.
...
-### Connection parameters in `config.toml`
-
-Add the following connection to your `config.toml`
-
-```toml
-[connections.integration]
-host = <host>
-account = <account_name>
-user = <user>
-password = <password>
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply CONTRIBUTING.md:81]: you an add the exports to your shell's RC file. long-term, I want to make the integration test runner read from this section, so i'll put it back in

File: tests_integration/scripts/integration_account_setup.sql [hunk 4]
```diff
@@ -73,15 +74,15 @@ CREATE OR REPLACE EXTERNAL ACCESS INTEGRATION snowflake_docs_access_integration
...
+CREATE API INTEGRATION IF NOT EXISTS snowcli_testing_repo_api_integration
 API_PROVIDER = git_https_api
 API_ALLOWED_PREFIXES = ('https://github.com/snowflakedb/')
 ALLOWED_AUTHENTICATION_SECRETS = ()
 ENABLED = true;
 GRANT USAGE ON INTEGRATION snowcli_testing_repo_api_integration TO ROLE INTEGRATION_TESTS;
 
 -- Notebooks setup
-CREATE DATABASE NOTEBOOK;
+CREATE DATABASE IF NOT EXISTS NOTEBOOK;
```
[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply tests_integration/scripts/integration_account_setup.sql:85]: i wasn't sure about this one, i assumed we wanted to create the DB but not `USE` it here

[the remark this answers is not in the document]
Comment (sfc-gh-melnacouzi) [reply tests_integration/scripts/integration_account_setup.sql:85]: I think it's needed, as there is Create Table after without mentioning of schema or database, and we need `create or replace table transcripts`.
Alternatively, we can change this to create or replace database, but not sure if it's safe or if that 'create or replace' command is available for db.

[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply tests_integration/scripts/integration_account_setup.sql:85]: @sfc-gh-turbaszek Given that the below queries are commented out, was the intention to have the `NOTEBOOK` database the active one at this point?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -65,44 +65,59 @@ or by running `pytest` inside activated environment.
...
+### Account setup
+The account used for integration tests must first be set up using the `ACCOUNTADMIN` role. Run
+```bash
+snow sql -c <connection name> -f tests_integration/scripts/integration_account_setup.sql
```
Comment (sfc-gh-astus) [inline CONTRIBUTING.md:73]: I'm working on this script, adding missing queries, fixing.

Comment (sfc-gh-fcampbell) [reply CONTRIBUTING.md:73]: ah cool, want me to hold off on this PR?

File: tests_integration/scripts/integration_account_setup.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfc-gh-astus) [reply tests_integration/scripts/integration_account_setup.sql:85]: In my PR I added `if not exists` to all queries and uncommented queries for cortex. PR should be ready tomorrow.

File: CONTRIBUTING.md (hunk 5 above)
Comment (sfc-gh-astus) [reply CONTRIBUTING.md:73]: No, you should merge it.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -65,44 +65,59 @@ or by running `pytest` inside activated environment.
...
-List of required parameter keys:
-- host
-- account
-- user
-- password
+The required environment variables are:
+```bash
+SNOWFLAKE_CONNECTIONS_INTEGRATION_HOST="<your host here>"
+SNOWFLAKE_CONNECTIONS_INTEGRATION_ACCOUNT="<your account here>"
+SNOWFLAKE_CONNECTIONS_INTEGRATION_USER="snowcli_test"
```
Comment (sfc-gh-bdufour) [inline CONTRIBUTING.md:98]: This is not configurable?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -65,44 +65,59 @@ or by running `pytest` inside activated environment.
...
+SNOWFLAKE_CONNECTIONS_INTEGRATION_PRIVATE_KEY_PATH="~/.ssh/snowcli_test_rsa_key.p8"
+SNOWFLAKE_CONNECTIONS_INTEGRATION_ROLE="integration_tests"
+SNOWFLAKE_CONNECTIONS_INTEGRATION_WAREHOUSE="xsmall"
```
Comment (sfc-gh-bdufour) [inline CONTRIBUTING.md:101]: These should be configurable too, if they are not currently.

File: CONTRIBUTING.md (hunk 6 above)
Comment (sfc-gh-fcampbell) [reply CONTRIBUTING.md:98]: it's hardcoded in the account setup script: https://github.com/snowflakedb/snowflake-cli/blob/285887170c5da55582b6a30c095fe3f66797ed1d/tests_integration/scripts/integration_account_setup.sql#L17

File: CONTRIBUTING.md (hunk 7 above)
Comment (sfc-gh-fcampbell) [reply CONTRIBUTING.md:101]: ditto: https://github.com/snowflakedb/snowflake-cli/blob/285887170c5da55582b6a30c095fe3f66797ed1d/tests_integration/scripts/integration_account_setup.sql#L33

**Comment by sfc-gh-fcampbell** [comment]
closing since these were updated in #1445

**Pull request closed without merging**